bank_statement_parse/parserlib.py

```diff
@@ -0,0 +1,228 @@
+# -*- encoding: utf-8 -*-
+"""Classes and definitions used in parsing bank statements."""
+##############################################################################
+#
+#  Copyright (C) 2015 Therp BV <http://therp.nl>.
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+class BankTransaction(dict):
+    """Single transaction that is part of a bank statement."""
+
+    @property
+    def value_date(self):
+        """property getter"""
+        return self['date']
+
+    @value_date.setter
+    def value_date(self, value_date):
+        """property setter"""
+        self['date'] = value_date
+
+    @property
+    def name(self):
+        """property getter"""
+        return self['name']
+
+    @name.setter
+    def name(self, name):
+        """property setter"""
+        self['name'] = name
+
+    @property
+    def transferred_amount(self):
+        """property getter"""
+        return self['amount']
+
+    @transferred_amount.setter
+    def transferred_amount(self, transferred_amount):
+        """property setter"""
+        self['amount'] = transferred_amount
+
+    @property
+    def eref(self):
+        """property getter"""
+        return self['ref']
+
+    @eref.setter
+    def eref(self, eref):
+        """property setter"""
+        self['ref'] = eref
+        if not self.message:
+            self.name = eref
+
+    @property
+    def message(self):
+        """property getter"""
+        return self._message
+
+    @message.setter
+    def message(self, message):
+        """property setter"""
+        self._message = message
+        self.name = message
+
+    @property
+    def remote_owner(self):
+        """property getter"""
+        return self['partner_name']
+
+    @remote_owner.setter
+    def remote_owner(self, remote_owner):
+        """property setter"""
+        self['partner_name'] = remote_owner
+        if not (self.message or self.eref):
+            self.name = remote_owner
+
+    @property
+    def remote_account(self):
+        """property getter"""
+        return self['account_number']
+
+    @remote_account.setter
+    def remote_account(self, remote_account):
+        """property setter"""
+        self['account_number'] = remote_account
+
```

Comment:
Can you add a ```note``` property? It's part of account.bank.statement.line and is displayed in the :information_source: icon in the reconciliation screen so it's useful to store all data found in the electronic statement that could not be mapped to other fields but may nevertheless be useful for humans to determine what the transaction is about.